frontend/components/product-list/sections/RelatedPostsSection/index.tsx

```diff
@@ -55,37 +53,20 @@ export function RelatedPostsSection({ tags = [] }: RelatedPostsSectionProps) {
    );
 }
 
-function usePosts(tags: string[]) {
-   const previousTags = usePrevious(tags);
-
-   const [state, setState] = useSafeSetState<{
-      posts: Post[];
-      isLoading: boolean;
-      error?: string;
-   }>({
-      posts: [],
-      isLoading: true,
-   });
+const computeRelatedPostsKey = (tags: string[]) => [
+   `related_posts?data=${encodeURIComponent(JSON.stringify({ tags }))}`,
+];
 
-   React.useEffect(() => {
-      if (!isEqual(tags, previousTags)) {
-         fetchPosts(tags)
-            .then((posts) => {
-               setState({
-                  posts,
-                  isLoading: false,
-                  error: undefined,
-               });
-            })
-            .catch((error) => {
-               setState((current) => ({
-                  ...current,
-                  isLoading: false,
-                  error: error.message,
-               }));
-            });
+function usePosts(tags: string[]) {
+   const query = useQuery(
+      computeRelatedPostsKey(tags),
+      async (): Promise<Post[]> => {
+         return await fetchPosts(tags);
+      },
```

Comment:
FYI: This `async` and `await` don't do anything useful because react-query just wants a promise-returning function and `fetchPosts` already does that.

This could be simplified like:
`() => fetchPosts(tags)`